src/canillita_news_handler.erl

```diff
@@ -0,0 +1,120 @@
+%%% @doc GET /news API handler
+-module(canillita_news_handler).
+
+-behaviour(trails_handler).
+-behaviour(lasse_handler).
+
+%% trails_handler behaviour callback
+-export([trails/0]).
+
+%% API
+-export([notify/1]).
+
+%% lasse_handler behaviour callbacks
+-export([ init/3
+        , handle_notify/2
+        , handle_info/2
+        , handle_error/3
+        , terminate/3
+        ]).
+
+-type event () :: {newsitem_created, #{}}.
+
+-type options() :: #{path => string()}.
+-type last_event_id() :: binary() | undefined.
+-type state() :: #{}.
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% trails_handler callback
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec trails() -> trails:trails().
+trails() ->
+  Metadata =
+    #{ get =>
+       #{ tags => ["news"]
+        , summary =>
+          "WARNING: Do not try to use this endpoint from this page."
+          " Swagger doesn't understand SSE"
+        , description =>
+          "Opens an [SSE] (http://www.w3.org/TR/eventsource/)"
+          " connection to retrieve news updates"
+        , externalDocs =>
+          #{ description => "RFC"
+           , url => "http://www.w3.org/TR/eventsource/"
+           }
+        , produces => ["application/json"]
+        }
+     },
+  Path = "/news",
+  Options = #{module => ?MODULE, init_args => #{path => Path}},
+  [trails:trail(Path, lasse_handler, Options, Metadata)].
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% API
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%% @doc sends an event to all the listeners
+-spec notify(Event::event()) -> ok.
+notify(Event) ->
+  lists:foreach(
+    fun(Listener) ->
+      lasse_handler:notify(Listener, Event)
+    end,
+    pg2:get_members(canillita_listeners)
+   ).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% lasse_handler callbacks
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec init( InitArgs::options()
+          , LastEventId::last_event_id()
+          , Req::cowboy_req:req()
+          ) ->
+  {ok, cowboy_req:req(), [lasse_handler:event()], state()}.
+init(_InitArgs, LastEventId, Req) ->
+  Req1 = sr_entities_handler:announce_req(Req, #{}),
+  NewsItems = case LastEventId of
+    undefined ->
+      canillita_newsitems_repo:fetch_all();
+    LastEventId ->
+      #{created_at := CreatedAt} = canillita_newsitems_repo:fetch(LastEventId),
+      canillita_newsitems_repo:fetch_all(CreatedAt)
```

Comment:
Instead of doing these 2 steps here, you should call `canillita_newsitems_repo:fetch_since(LastEventId)` and do the appropriate computations inside the repo. That way, if we ever change the backend/store or improve the algorithm to find the events somehow, we only have to change the repo and not the handlers that use its functions.